src/Estimators/include/BipedalLocomotionControllers/Estimators/RecursiveLeastSquare.h

```diff
@@ -0,0 +1,118 @@
+
+    iDynTree::MatrixDynSize m_stateCovarianceMatrix; /**< Covariance matrix of the state */
+
+    /** Covariance matrix of the measuraments we assume that the measuraments are uncorrelated
+     (furthermore since the model of the noise is Gaussian the random variable are also independent)
+     Since the variable are independent we are interested only on the element in the diagonal
+     of the matrix */
+    iDynTree::MatrixDynSize m_measurementCovarianceMatrix;
+
+    iDynTree::MatrixDynSize m_kalmanGain; /**< Gain of the Kalman filter */
+
+    double m_lambda; /**< Filter gain. The recursive least square filter is equivalent to a kalman
+                        filter if lambda is equal to 1 */
+
+    std::function<iDynTree::MatrixDynSize(void)> m_regressor; /**< Function containing the regressor
+                                                                 of the system */
+
+    /**
+     * Enumerator useful to described the current status of the filter
+     */
+    enum class State
+    {
+        NotInitialized, /**< The filter is not initialized yet please class
+                           RecursiveLeastSquare::initialze */
+        Initialized, /**< The filter is initialized and ready to be used */
+        Running /**< The filter is running */
+    };
+
+    State m_estimatorState{State::NotInitialized}; /**< State of the filter */
+
+public:
+    /**
+     * Initialize the filter
+     * @note The following parameter are required by the filter:
+     * - "measurement_covariance" vector containing the covariance of the measuraments. We assume
+     that the measuraments are uncorrelated (furthermore since the model of the noise is Gaussian
+     the random variable are also independent) Since the variable are independent we are interested
+     only on the element in the diagonal of the matrix
+     * - "lambda" double containing the filter gain. (The recursive least square filter is
+     equivalent to a kalman filter if lambda is equal to 1)
+     * - "state" vector containing the initial guess of the state
+     * - "state_covariance" vector containing the diagonal matrix of the covariance of the state
+     * @param handlerWeak weak pointer to a ParametersHandler::IParametersHandler interface
+     * @tparameter Derived particular implementation of the IParameterHandler
+     * @return true in case of success, false otherwise
+     */
+    template <class Derived>
+    bool initialize(std::weak_ptr<ParametersHandler::IParametersHandler<Derived>> handlerWeak);
+
+    /**
+     * Set the regressor
+     * @parameter regressor fucntion that return an iDynTree::MatrixDynSize containing the regressor
+     * of the system
+     */
+    void setRegressorFunction(std::function<iDynTree::MatrixDynSize(void)> regressor);
+
+    /**
+     * Set the measuraments
+     * @parameter measurements vector containing all the measuraments
+     */
+    void setMeasuraments(const iDynTree::VectorDynSize& measuraments);
```

Comment:
setMeasurements

Reply:
Ops!

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/23/commits/3d85fca563f50d93670b5c4e67301fcbcace0ce2